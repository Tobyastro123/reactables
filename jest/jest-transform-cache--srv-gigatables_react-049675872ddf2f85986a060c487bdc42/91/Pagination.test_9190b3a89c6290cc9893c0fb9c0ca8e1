// import React from 'react'
// import classNames from 'classnames/bind'
// import styles from '../css/styles.css'
//
// var CommonConstants = require('./CommonConstants');
// var EditorConstants = require('./EditorConstants');
// var Lang = require('./Lang');

import Pagination from './src/components/Pagination.js';

var pgn = new Pagination({ page: 1 });

console.log(pgn);

// test('creates Pagination component')