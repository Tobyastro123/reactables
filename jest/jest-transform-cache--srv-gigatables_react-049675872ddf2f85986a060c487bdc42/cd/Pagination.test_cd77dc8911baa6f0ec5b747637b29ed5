/* istanbul ignore next */'use strict';







var /* istanbul ignore next */_Pagination = require('./src/components/Pagination.js'); /* istanbul ignore next */var _Pagination2 = _interopRequireDefault(_Pagination);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

var pgn = new /* istanbul ignore next */_Pagination2.default({ page: 1 }); // import React from 'react'
// import classNames from 'classnames/bind'
// import styles from '../css/styles.css'
//
// var CommonConstants = require('./CommonConstants');
// var EditorConstants = require('./EditorConstants');
// var Lang = require('./Lang');
console.log(pgn); // test('creates Pagination component')