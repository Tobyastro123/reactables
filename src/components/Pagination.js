import React from 'react'
import classNames from 'classnames/bind';
import styles from '../css/styles.css'

var CommonConstants = require('./CommonConstants');
var EditorConstants = require('./EditorConstants');
var Lang = require('./Lang');

class Pagination extends React.Component {
  render()
  {
    var lang = Lang[this.props.lang];
    let prevClasses = classNames({
      gt_page: true,
      prev: true
    });
    let nextClasses = classNames({
      gt_page: true,
      next: true
    });

    let countRows = this.props.countRows,
    page = this.props.page,
    perPage = this.props.perPage,
    pages = Math.ceil(countRows / perPage),
    more = 5;

    let pagesContent = [];
    for (var i=0;i<pages;++i) {
      let currentPage = i+1;
      let pageClasses = classNames({
        gt_page: true,
        selected: (currentPage === page)?true:false
      });
      pagesContent[i] = <div key={i} onClick={this.props.updatePagination}
      data-from={i*perPage} className={pageClasses}>{currentPage}</div>;
    }

    let prev = (page === 1) ? perPage * (pages - 1) : perPage * (page - 2),
    next = (page === pages) ? 0 : perPage * (page);

    let from = parseInt(this.props.fromRow);
    let description = lang.showing + ' ' + (from + 1) + ' '
    + lang.to + ' ' + (page * perPage) + ' ' + lang.of  + ' ' + countRows + ' ' + lang.entries + '.';
    return (
      <div className={styles.gt_pagination}>
        <div className={styles.gt_pgn_ttl}>{description}</div>
        <div className={styles.gt_pgn_pages}>
          <div className={styles.gt_pagn}>
            <div data-from={prev} onClick={this.props.updatePagination} className={prevClasses}>{lang.prev}</div>
            {pagesContent}
            <div data-from={next} onClick={this.props.updatePagination} className={nextClasses}>{lang.next}</div>
          </div>
        </div>
        <div className={styles.clear}></div>
      </div>
    )
  }
}

export default Pagination
