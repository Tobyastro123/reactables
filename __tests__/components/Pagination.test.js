import React from 'react';
import renderer from 'react-test-renderer';
import Pagination from '../../src/components/Pagination.js'
import { shallow } from 'enzyme';

it('renders Pagination correctly', () => {
  const tree = renderer.create(
    <Pagination
    lang="en"
    countRows={249}
    fromRow={0}
    page={1}
    perPage={50} />
  ).toJSON();
  expect(tree).toMatchSnapshot();

  const treeMore = renderer.create(
    <Pagination
    lang="en"
    countRows={555}
    fromRow={450}
    page={9}
    perPage={50} />
  ).toJSON();
  expect(treeMore).toMatchSnapshot();

  const obj = shallow(
    <Pagination
    lang="en"
    countRows={229}
    fromRow={0}
    page={1}
    perPage={50} />
  );

  obj.instance().shouldComponentUpdate({
    lang: 'en',
    countRows: 230,
    fromRow: 0,
    page: 1,
    perPage: 50
  });
});
