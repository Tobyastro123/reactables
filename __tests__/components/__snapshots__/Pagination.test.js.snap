exports[`test renders Pagination correctly 1`] = `
<div
  className={undefined}>
  <div
    className={undefined}>
    Showing 1 to 50 of 229 entries.
  </div>
  <div
    className={undefined}>
    <div
      className={undefined}>
      <div
        className="gt_page prev"
        data-from={200}
        onClick={undefined}>
        Previous
      </div>
      <div
        className="gt_page selected"
        data-from={0}
        onClick={undefined}>
        1
      </div>
      <div
        className="gt_page"
        data-from={50}
        onClick={undefined}>
        2
      </div>
      <div
        className="gt_page"
        data-from={100}
        onClick={undefined}>
        3
      </div>
      <div
        className="gt_page"
        data-from={150}
        onClick={undefined}>
        4
      </div>
      <div
        className="gt_page"
        data-from={200}
        onClick={undefined}>
        5
      </div>
      <div
        className="gt_page next"
        data-from={50}
        onClick={undefined}>
        Next
      </div>
    </div>
  </div>
  <div
    className={undefined} />
</div>
`;
